### PR TITLE
ignore UTF-8 BOM

### LINKED ISSUE #592: protoc doesn't support files starting with Byte order marks.
Trying to compile .proto file starting with unicode byte order mark ( U+FEFF character) results in an error.

This is extremely painful in Visual Studio, which saves all files with a byteorder mark, so basically every .proto file that you write in Visual Studio editor will be considered broken by protoc.

The error message is also not very helpful:
health.proto:1:1: Interpreting non ascii codepoint 239.
[libprotobuf WARNING google/protobuf/compiler/parser.cc:491] No syntax specified
 for the proto file. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to
specify a syntax version. (Defaulted to proto2 syntax.)
health.proto:1:1: Expected top-level statement (e.g. "message").
health.proto:1:2: Interpreting non ascii codepoint 187.
health.proto:1:3: Interpreting non ascii codepoint 191.

### PR DESCRIPTION
**Pull request by anandolee**
fix #592

@jtattermusch @xfxyjwf

### CODE REVIEW

**Comment by xfxyjwf** [comment]
Add test cases?

**Comment by jtattermusch** [comment]
+1 to adding a test(s)

File: src/google/protobuf/io/tokenizer.cc [hunk 1]
```diff
@@ -762,6 +762,13 @@ bool Tokenizer::NextWithComments(string* prev_trailing_comments,
                              next_leading_comments);
 
   if (current_.type == TYPE_START) {
+    if (TryConsume((char)0xEF)) {
```
Comment (jtattermusch) [inline src/google/protobuf/io/tokenizer.cc]: Proto file start -> Proto file starts

Comment (jtattermusch) [inline src/google/protobuf/io/tokenizer.cc]: Perhaps add a comment saying this adds support for reading UTF-8 BOM 0xFE, 0xBB, 0xBF prologue.

**Comment by anandolee** [comment]
Tests added

**Comment by jtattermusch** [comment]
LGTM, good to merge after travis & appveyor are green.

**Comment by xfxyjwf** [comment]
LGTM

**Pull request merged**